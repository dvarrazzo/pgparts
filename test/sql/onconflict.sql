\pset format unaligned
\pset footer off
\set VERBOSITY terse
create schema pconf;
set datestyle = 'ISO';

create extension pgparts with schema pconf;


-- No drop on conflict

create table conftbl0 (
    id serial primary key,
    day date not null,
    data text unique);

select pconf.setup('conftbl0', 'day', 'monthly',
    '{{nmonths,3},{on_conflict_drop,false}}');
select pconf.create_for('conftbl0'::regclass, '2015-12-10');

insert into conftbl0 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl0 (id, day, data) values (10, '2015-12-11', 'bar');
insert into conftbl0 (id, day, data) values (11, '2015-12-12', 'foo');

select tableoid::regclass, * from conftbl0 order by id;


-- Drop on conflict on montly schema

create table conftbl1 (
    id serial primary key,
    day date not null,
    data text unique);

select pconf.setup('conftbl1', 'day', 'monthly',
    '{{nmonths,3},{on_conflict_drop,true}}');
select pconf.create_for('conftbl1'::regclass, '2015-12-10');

insert into conftbl1 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl1 (id, day, data) values (10, '2015-12-11', 'bar');
insert into conftbl1 (id, day, data) values (11, '2015-12-12', 'foo');

select tableoid::regclass, * from conftbl1 order by id;


-- Drop on conflict on daily schema

create table conftbl2 (
    id serial primary key,
    day date not null,
    data text unique);

select pconf.setup('conftbl2', 'day', 'daily',
    '{{ndays,7},{on_conflict_drop,true}}');
select pconf.create_for('conftbl2'::regclass, '2015-12-10');

insert into conftbl2 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl2 (id, day, data) values (10, '2015-12-11', 'bar');
insert into conftbl2 (id, day, data) values (11, '2015-12-12', 'foo');

select tableoid::regclass, * from conftbl2 order by id;


drop extension pgparts;
