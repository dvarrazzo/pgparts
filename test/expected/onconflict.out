\pset format unaligned
\pset footer off
\set VERBOSITY terse
create schema pconf;
set datestyle = 'ISO';
create extension pgparts with schema pconf;
-- No drop on conflict
create table conftbl0 (
    id serial primary key,
    day date not null,
    data text unique);
select pconf.setup('conftbl0', 'day', 'monthly',
    '{{nmonths,3},{on_conflict_drop,false}}');
setup

select pconf.create_for('conftbl0'::regclass, '2015-12-10');
NOTICE:  creating partition public.conftbl0_201510 of table conftbl0
create_for
conftbl0_201510
insert into conftbl0 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl0 (id, day, data) values (10, '2015-12-11', 'bar');
ERROR:  duplicate key value violates unique constraint "conftbl0_201510_pkey"
insert into conftbl0 (id, day, data) values (11, '2015-12-12', 'foo');
ERROR:  duplicate key value violates unique constraint "conftbl0_201510_data_key"
select tableoid::regclass, * from conftbl0 order by id;
tableoid|id|day|data
conftbl0_201510|10|2015-12-10|foo
-- Drop on conflict on montly schema
create table conftbl1 (
    id serial primary key,
    day date not null,
    data text unique);
select pconf.setup('conftbl1', 'day', 'monthly',
    '{{nmonths,3},{on_conflict_drop,true}}');
setup

select pconf.create_for('conftbl1'::regclass, '2015-12-10');
NOTICE:  creating partition public.conftbl1_201510 of table conftbl1
create_for
conftbl1_201510
insert into conftbl1 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl1 (id, day, data) values (10, '2015-12-11', 'bar');
insert into conftbl1 (id, day, data) values (11, '2015-12-12', 'foo');
ERROR:  duplicate key value violates unique constraint "conftbl1_201510_data_key"
select tableoid::regclass, * from conftbl1 order by id;
tableoid|id|day|data
conftbl1_201510|10|2015-12-10|foo
-- Drop on conflict on daily schema
create table conftbl2 (
    id serial primary key,
    day date not null,
    data text unique);
select pconf.setup('conftbl2', 'day', 'daily',
    '{{ndays,7},{on_conflict_drop,true}}');
setup

select pconf.create_for('conftbl2'::regclass, '2015-12-10');
NOTICE:  creating partition public.conftbl2_20151206 of table conftbl2
create_for
conftbl2_20151206
insert into conftbl2 (id, day, data) values (10, '2015-12-10', 'foo');
insert into conftbl2 (id, day, data) values (10, '2015-12-11', 'bar');
insert into conftbl2 (id, day, data) values (11, '2015-12-12', 'foo');
ERROR:  duplicate key value violates unique constraint "conftbl2_20151206_data_key"
select tableoid::regclass, * from conftbl2 order by id;
tableoid|id|day|data
conftbl2_20151206|10|2015-12-10|foo
drop extension pgparts;
